ios/PurchasesHybridCommon/PurchasesHybridCommon/PaywallProxy.swift

```diff
@@ -30,50 +32,107 @@ import UIKit
         return PaywallFooterViewController()
     }
 
+    @available(*, deprecated, message: "Use presentPaywall with onPaywallResult instead")
     @objc
     public func presentPaywall() {
-        privatePresentPaywall(displayCloseButton: nil, offering: nil)
+        self.privatePresentPaywall(displayCloseButton: nil, offering: nil)
     }
 
+    @available(*, deprecated, message: "Use presentPaywall with onPaywallResult instead")
     @objc
     public func presentPaywall(displayCloseButton: Bool) {
-        privatePresentPaywall(displayCloseButton: displayCloseButton, offering: nil)
+        self.privatePresentPaywall(displayCloseButton: displayCloseButton, offering: nil)
     }
 
+    @available(*, deprecated, message: "Use presentPaywall with onPaywallResult instead")
     @objc
     public func presentPaywall(offering: Offering) {
-        privatePresentPaywall(displayCloseButton: nil, offering: offering)
+        self.privatePresentPaywall(displayCloseButton: nil, offering: offering)
     }
 
+    @available(*, deprecated, message: "Use presentPaywall with onPaywallResult instead")
     @objc
     public func presentPaywall(offering: Offering, displayCloseButton: Bool) {
-        privatePresentPaywall(displayCloseButton: displayCloseButton, offering: offering)
+        self.privatePresentPaywall(displayCloseButton: displayCloseButton, offering: offering)
+    }
+
+    @objc
+    public func presentPaywall(onPaywallResult: @escaping (String) -> Void) {
```

Comment:
this name on a parameter feels very android-y, on iOS I'd have expected it to be called something like `paywallResultHandler`

Reply:
Made the rename in https://github.com/RevenueCat/purchases-hybrid-common/pull/628/commits/2551cf217ae977773950ed0e9a6c5bea4f53c113